src/routes/v1/secret_test.go

```diff
@@ -11,76 +12,376 @@ import (
-
-	writer := httptest.NewRecorder()
-	router.ServeHTTP(writer, request)
-
-	assert.Equal(t, http.StatusOK, writer.Code)
-	var response types.CreateSecretResponse
-	err := json.Unmarshal(writer.Body.Bytes(), &response)
-	assert.NoError(t, err)
-	assert.Equal(t, "new-secret", response.SecretName)
-	assert.Equal(t, "default", response.NamespaceName)
+func TestGetSecret(t *testing.T) {
+	secretNamespace := "test-namespace-secret"
+	secretName := "test-secret"
+	type requestUri struct {
+		namespace string
+		secret    string
+	}
+
+	type want struct {
+		statusCode int
+		response   map[string]interface{}
+	}
+
+	cases := map[string]struct {
+		requestUri requestUri
+		want       want
+	}{
+		"ShouldSuccessGettingSecret": {
+			requestUri: requestUri{
+				secret:    secretName,
+				namespace: secretNamespace,
+			},
+			want: want{
+				statusCode: http.StatusOK,
+				response: map[string]interface{}{
+					"secretName": secretName,
+					"type":       "Opaque",
+					"id":         "",
+					"data":       []interface{}{map[string]interface{}{"key": "key1", "value": "fake"}},
+				},
+			},
+		},
+		"ShouldNotFoundSecret": {
+			requestUri: requestUri{
+				secret:    "test-not-exists",
+				namespace: secretNamespace,
+			},
+			want: want{
+				statusCode: http.StatusNotFound,
+				response: map[string]interface{}{
+					"details": fmt.Sprintf("secrets %q not found", "test-not-exists"),
+					"error":   "Operation failed",
+				},
+			},
+		},
+		"ShouldNotFoundNamespace": {
+			requestUri: requestUri{
+				secret:    secretName,
+				namespace: "test-not-exists",
+			},
+			want: want{
+				statusCode: http.StatusNotFound,
+				response: map[string]interface{}{
+					"details": fmt.Sprintf("secrets %q not found", secretName),
+					"error":   "Operation failed",
+				},
+			},
+		},
+	}
+	createTestNamespace(secretNamespace)
+	createTestSecret(secretName, secretNamespace)
+
+	for name, test := range cases {
+		t.Run(name, func(t *testing.T) {
+			request, _ := http.NewRequest("GET", fmt.Sprintf("/v1/namespaces/%s/secrets/%s",
+				test.requestUri.namespace, test.requestUri.secret), nil)
+			writer := httptest.NewRecorder()
+			router.ServeHTTP(writer, request)
+			assert.Equal(t, test.want.statusCode, writer.Code)
+
+			var response map[string]interface{}
+			if err := json.Unmarshal(writer.Body.Bytes(), &response); err != nil {
+				panic(err)
+			}
+			assert.Equal(t, test.want.response, response)
+
+		})
+	}
 }
 
 func TestGetSecrets(t *testing.T) {
-	request, _ := http.NewRequest("GET", "/v1/namespaces/default/secrets/", nil)
-	writer := httptest.NewRecorder()
-	router.ServeHTTP(writer, request)
-
-	assert.Equal(t, http.StatusOK, writer.Code)
-	var response types.GetSecretsResponse
-	err := json.Unmarshal(writer.Body.Bytes(), &response)
-	assert.NoError(t, err)
+	secretNamespace := "test-namespace-secrets"
+	type requestUri struct {
+		namespace string
+	}
+
+	type want struct {
+		statusCode int
+		response   types.GetSecretsResponse
```

Comment:
Change the response to `map[string]interface{}` so it looks like the other tests